tests/mlmodel_sklearn/test_cluster_models.py

```diff
@@ -0,0 +1,186 @@
+        ],
+        "SpectralBiclustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralBiclustering.fit", 1),
+        ],
+        "SpectralCoclustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralCoclustering.fit", 1),
+        ],
+        "SpectralClustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralClustering.fit", 2),
+            ("Function/MLModel/Sklearn/Named/SpectralClustering.fit_predict", 1),
+        ],
+    }
+    expected_transaction_name = "test_cluster_models:_test"
+    if six.PY3:
+        expected_transaction_name = (
+            "test_cluster_models:test_below_v1_1_model_methods_wrapped_in_function_trace.<locals>._test"
+        )
+
+    @validate_transaction_metrics(
+        expected_transaction_name,
+        scoped_metrics=expected_scoped_metrics[cluster_model_name],
+        rollup_metrics=expected_scoped_metrics[cluster_model_name],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        run_cluster_model(cluster_model_name)
+
+    _test()
+
+
+@pytest.mark.skipif(SKLEARN_VERSION < (1, 1, 0), reason="Requires sklearn > 1.1")
+@pytest.mark.parametrize(
+    "cluster_model_name",
+    [
+        "BisectingKMeans",
+        "OPTICS",
+    ],
+)
+def test_above_v1_1_model_methods_wrapped_in_function_trace(cluster_model_name, run_cluster_model):
+    expected_scoped_metrics = {
+        "BisectingKMeans": [
+            ("Function/MLModel/Sklearn/Named/BisectingKMeans.fit", 2),
+            ("Function/MLModel/Sklearn/Named/BisectingKMeans.predict", 1),
+            ("Function/MLModel/Sklearn/Named/BisectingKMeans.fit_predict", 1),
+        ],
+        "OPTICS": [
+            ("Function/MLModel/Sklearn/Named/OPTICS.fit", 2),
+            ("Function/MLModel/Sklearn/Named/OPTICS.fit_predict", 1),
+        ],
+    }
+    expected_transaction_name = "test_cluster_models:_test"
+    if six.PY3:
+        expected_transaction_name = (
+            "test_cluster_models:test_above_v1_1_model_methods_wrapped_in_function_trace.<locals>._test"
+        )
+
+    @validate_transaction_metrics(
+        expected_transaction_name,
+        scoped_metrics=expected_scoped_metrics[cluster_model_name],
+        rollup_metrics=expected_scoped_metrics[cluster_model_name],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        run_cluster_model(cluster_model_name)
+
+    _test()
+
+
+@pytest.fixture
+def run_cluster_model():
+    def _run(cluster_model_name):
+        import sklearn.cluster
+        from sklearn.datasets import load_iris
+        from sklearn.model_selection import train_test_split
+
+        X, y = load_iris(return_X_y=True)
+        x_train, x_test, y_train, y_test = train_test_split(X, y, stratify=y, random_state=0)
+
+        clf = getattr(sklearn.cluster, cluster_model_name)()
+
+        model = clf.fit(x_train, y_train)
+
+        if hasattr(model, "predict"):
```

Comment:
Assuming we don't want to include classes that don't have a predict method you can remove this hasattr check around the predict.

Reply:
Since we are instrumenting models that have fit (and not necessarily fit and predict) we might run across models that have only "fit"